core/primitives/src/block_header.rs

```diff
@@ -563,6 +719,16 @@ impl BlockHeader {
         match self {
             BlockHeader::BlockHeaderV1(header) => &header.inner_rest.chunk_mask,
             BlockHeader::BlockHeaderV2(header) => &header.inner_rest.chunk_mask,
+            BlockHeader::BlockHeaderV3(header) => &header.inner_rest.chunk_mask,
+        }
+    }
+
+    #[inline]
+    pub fn block_ordinal(&self) -> NumBlocks {
+        match self {
+            BlockHeader::BlockHeaderV1(_) => 0, // not applicable
```

Comment:
Should we return an option then? It is better to take advantage of the rust type system than to rely on some hardcoded constant.

Reply:
We can do that.
However, BlockHeader is very basic primitive that is used everywhere in the code. Once we use Option<NumBlocks> for small piece of semantical sugar here, we will have to unwrap it everywhere for forever. I wouldn't go this way as it may create tons of `match ...` and `unwrap()` code later with no actual purpose.

Reply:
We can introduce Option wrapper for block_ordinal in BlockHeaderView only. It makes sense to me. Should we go this way?

Reply:
Yeah let's do that